lc2fen/infer_pieces.py

```diff
@@ -272,6 +285,257 @@ def __generate_fen_based_on_previous_fen_and_detected_move(
         return previous_board.board_fen()
 
 
+def __determine_most_probable_white_piece(pieces_probs, square):
+    """Determines the most probable white piece."""
+    most_probable_piece_prob = 0
+    if pieces_probs[square][4] > most_probable_piece_prob:
+        most_probable_piece = "Q"
+        most_probable_piece_prob = pieces_probs[square][4]
+    if pieces_probs[square][2] > most_probable_piece_prob:
+        most_probable_piece = "N"
+        most_probable_piece_prob = pieces_probs[square][2]
+    if pieces_probs[square][5] > most_probable_piece_prob:
+        most_probable_piece = "R"
+        most_probable_piece_prob = pieces_probs[square][5]
+    if pieces_probs[square][0] > most_probable_piece_prob:
+        most_probable_piece = "B"
+        most_probable_piece_prob = pieces_probs[square][0]
+    if pieces_probs[square][3] > most_probable_piece_prob:
+        most_probable_piece = "P"
+    return most_probable_piece
+
+
+def __determine_most_probable_black_piece(pieces_probs, square):
+    """Determines the most probable black piece."""
+    most_probable_piece_prob = 0
+    if pieces_probs[square][11] > most_probable_piece_prob:
+        most_probable_piece = "q"
+        most_probable_piece_prob = pieces_probs[square][4]
+    if pieces_probs[square][9] > most_probable_piece_prob:
+        most_probable_piece = "n"
+        most_probable_piece_prob = pieces_probs[square][2]
+    if pieces_probs[square][12] > most_probable_piece_prob:
+        most_probable_piece = "r"
+        most_probable_piece_prob = pieces_probs[square][5]
+    if pieces_probs[square][7] > most_probable_piece_prob:
+        most_probable_piece = "b"
+        most_probable_piece_prob = pieces_probs[square][0]
+    if pieces_probs[square][10] > most_probable_piece_prob:
+        most_probable_piece = "p"
+    return most_probable_piece
+
+
+def __check_balance_among_pawns_queens_and_bishops(
```

Comment:
This is a great idea! However, I believe this could be simplified significantly if we use additions instead of if/elif for the conditions. For example, if we want to add a `P, B, or Q`, we want to check that `P+(Q-1)+(B_dark_squared-1)+(B_light_squared-1) <= 8` (or some idea like this, don't take it literally because I haven't tested it). Could you give it a try? :)

Reply:
Thanks and yes! I've reimplemented the function using your suggestion